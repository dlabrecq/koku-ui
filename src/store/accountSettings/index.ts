import * as accountSettingsActions from './accountSettingsActions';
import { stateKey as accountSettingsStateKey } from './accountSettingsCommon';
import { AccountSettingsAction, accountSettingsReducer, AccountSettingsState } from './accountSettingsReducer';
import * as accountSettingsSelectors from './accountSettingsSelectors';

export {
  AccountSettingsAction,
  accountSettingsActions,
  accountSettingsReducer,
  accountSettingsSelectors,
  AccountSettingsState,
  accountSettingsStateKey,
};
