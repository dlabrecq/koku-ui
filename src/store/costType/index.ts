import * as costTypeActions from './costTypeActions';
import { stateKey as costTypeStateKey } from './costTypeCommon';
import { CostTypeAction, costTypeReducer, CostTypeState } from './costTypeReducer';
import * as costTypeSelectors from './costTypeSelectors';

export { CostTypeAction, costTypeActions, costTypeReducer, costTypeSelectors, CostTypeState, costTypeStateKey };
