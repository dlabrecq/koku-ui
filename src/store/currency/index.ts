import * as currencyActions from './currencyActions';
import { stateKey as currencyStateKey } from './currencyCommon';
import { CurrencyAction, currencyReducer, CurrencyState } from './currencyReducer';
import * as currencySelectors from './currencySelectors';

export { CurrencyAction, currencyActions, currencyReducer, currencySelectors, CurrencyState, currencyStateKey };
