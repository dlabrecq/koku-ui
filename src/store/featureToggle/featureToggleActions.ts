import { createAction } from 'typesafe-actions';

export interface FeatureToggleActionMeta {
  isClusterInfoToggleEnabled?: boolean;
  isDebugToggleEnabled?: boolean;
  isExportsToggleEnabled?: boolean;
  isFinsightsToggleEnabled?: boolean;
  isIbmToggleEnabled?: boolean;
  isRosToggleEnabled?: boolean;
  isSettingsPlatformToggleEnabled?: boolean;
  isTagMappingToggleEnabled?: boolean;
}

export const setFeatureToggle = createAction('feature/init_feature_toggle')<FeatureToggleActionMeta>();
export const resetState = createAction('feature/reset_state')();
