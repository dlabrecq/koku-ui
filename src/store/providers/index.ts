import {
  awsProvidersQuery,
  azureProvidersQuery,
  gcpProvidersQuery,
  ibmProvidersQuery,
  ocpProvidersQuery,
  providersQuery,
} from 'store/providers/providersCommon';

import * as providersActions from './providersActions';
import { stateKey as providersStateKey } from './providersCommon';
import { ProvidersAction, providersReducer, ProvidersState } from './providersReducer';
import * as providersSelectors from './providersSelectors';

export {
  awsProvidersQuery,
  azureProvidersQuery,
  gcpProvidersQuery,
  ibmProvidersQuery,
  ocpProvidersQuery,
  ProvidersAction,
  providersActions,
  providersQuery,
  providersReducer,
  providersSelectors,
  ProvidersState,
  providersStateKey,
};
