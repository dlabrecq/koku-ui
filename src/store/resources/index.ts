import * as resourceActions from './resourceActions';
import { resourceStateKey } from './resourceCommon';
import { CachedResource, ResourceAction, resourceReducer, ResourceState } from './resourceReducer';
import * as resourceSelectors from './resourceSelectors';

export {
  ResourceAction,
  CachedResource,
  resourceActions,
  resourceReducer,
  resourceSelectors,
  ResourceState,
  resourceStateKey,
};
