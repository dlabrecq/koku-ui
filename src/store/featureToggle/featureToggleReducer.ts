import type { ActionType } from 'typesafe-actions';
import { getType } from 'typesafe-actions';

import type { resetState } from './featureToggleActions';
import { setFeatureToggle } from './featureToggleActions';

export type FeatureToggleAction = ActionType<typeof setFeatureToggle | typeof resetState>;

export type FeatureToggleState = Readonly<{
  hasFeatureToggle: boolean;
  isClusterInfoToggleEnabled: boolean;
  isDebugToggleEnabled: boolean;
  isExportsToggleEnabled: boolean;
  isFinsightsToggleEnabled: boolean;
  isIbmToggleEnabled: boolean;
  isRosToggleEnabled: boolean;
  isSettingsPlatformToggleEnabled: boolean;
  isTagMappingToggleEnabled: boolean;
}>;

export const defaultState: FeatureToggleState = {
  hasFeatureToggle: false,
  isClusterInfoToggleEnabled: false,
  isDebugToggleEnabled: false,
  isExportsToggleEnabled: false,
  isFinsightsToggleEnabled: false,
  isIbmToggleEnabled: false,
  isRosToggleEnabled: false,
  isSettingsPlatformToggleEnabled: false,
  isTagMappingToggleEnabled: false,
};

export const stateKey = 'FeatureToggle';

export function FeatureToggleReducer(state = defaultState, action: FeatureToggleAction): FeatureToggleState {
  switch (action.type) {
    case getType(setFeatureToggle):
      return {
        ...state,
        hasFeatureToggle: true,
        isClusterInfoToggleEnabled: action.payload.isClusterInfoToggleEnabled,
        isDebugToggleEnabled: action.payload.isDebugToggleEnabled,
        isExportsToggleEnabled: action.payload.isExportsToggleEnabled,
        isFinsightsToggleEnabled: action.payload.isFinsightsToggleEnabled,
        isIbmToggleEnabled: action.payload.isIbmToggleEnabled,
        isRosToggleEnabled: action.payload.isRosToggleEnabled,
        isSettingsPlatformToggleEnabled: action.payload.isSettingsPlatformToggleEnabled,
        isTagMappingToggleEnabled: action.payload.isTagMappingToggleEnabled,
      };

    default:
      return state;
  }
}
