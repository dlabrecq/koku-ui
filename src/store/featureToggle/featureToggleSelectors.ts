import type { RootState } from 'store/rootReducer';

import { stateKey } from './featureToggleReducer';

export const selectFeatureToggleState = (state: RootState) => state[stateKey];

export const selectHasFeatureToggle = (state: RootState) => selectFeatureToggleState(state).hasFeatureToggle;

export const selectIsAwsEc2InstancesToggleEnabled = (state: RootState) =>
  selectFeatureToggleState(state).isAwsEc2InstancesToggleEnabled;
export const selectIsDebugToggleEnabled = (state: RootState) => selectFeatureToggleState(state).isDebugToggleEnabled;
export const selectIsExportsToggleEnabled = (state: RootState) =>
  selectFeatureToggleState(state).isExportsToggleEnabled;
export const selectIsFinsightsToggleEnabled = (state: RootState) =>
  selectFeatureToggleState(state).isFinsightsToggleEnabled;
export const selectIsIbmToggleEnabled = (state: RootState) => selectFeatureToggleState(state).isIbmToggleEnabled;
export const selectIsOcpCloudNetworkingToggleEnabled = (state: RootState) =>
  selectFeatureToggleState(state).isOcpCloudNetworkingToggleEnabled;
export const selectIsOcpProjectStorageToggleEnabled = (state: RootState) =>
  selectFeatureToggleState(state).isOcpProjectStorageToggleEnabled;
export const selectIsRosToggleEnabled = (state: RootState) => selectFeatureToggleState(state).isRosToggleEnabled;
