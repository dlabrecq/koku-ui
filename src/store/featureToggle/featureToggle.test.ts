import { createMockStoreCreator } from 'store/mockStore';

import { FeatureToggleSelectors } from '.';
import * as actions from './featureToggleActions';
import { FeatureToggleReducer, stateKey } from './featureToggleReducer';
import * as selectors from './featureToggleSelectors';

const createUIStore = createMockStoreCreator({
  [stateKey]: FeatureToggleReducer,
});

test('default state', async () => {
  const store = createUIStore();
  expect(selectors.selectFeatureToggleState(store.getState())).toMatchSnapshot();
});

test('AWS EC2 instances feature is enabled', async () => {
  const store = createUIStore();
  store.dispatch(actions.setFeatureToggle({ isAwsEc2InstancesToggleEnabled: true }));
  expect(FeatureToggleSelectors.selectIsAwsEc2InstancesToggleEnabled(store.getState())).toBe(true);
});

test('Debug feature is enabled', async () => {
  const store = createUIStore();
  store.dispatch(actions.setFeatureToggle({ isDebugToggleEnabled: true }));
  expect(FeatureToggleSelectors.selectIsDebugToggleEnabled(store.getState())).toBe(true);
});

test('Exports feature is enabled', async () => {
  const store = createUIStore();
  store.dispatch(actions.setFeatureToggle({ isExportsToggleEnabled: true }));
  expect(FeatureToggleSelectors.selectIsExportsToggleEnabled(store.getState())).toBe(true);
});

test('FINsights feature is enabled', async () => {
  const store = createUIStore();
  store.dispatch(actions.setFeatureToggle({ isFinsightsToggleEnabled: true }));
  expect(FeatureToggleSelectors.selectIsFinsightsToggleEnabled(store.getState())).toBe(true);
});

test('IBM feature is enabled', async () => {
  const store = createUIStore();
  store.dispatch(actions.setFeatureToggle({ isIbmToggleEnabled: true }));
  expect(FeatureToggleSelectors.selectIsIbmToggleEnabled(store.getState())).toBe(true);
});

test('OCP on cloud networking feature is enabled', async () => {
  const store = createUIStore();
  store.dispatch(actions.setFeatureToggle({ isOcpCloudNetworkingToggleEnabled: true }));
  expect(FeatureToggleSelectors.selectIsOcpCloudNetworkingToggleEnabled(store.getState())).toBe(true);
});

test('OCP project storage feature is enabled', async () => {
  const store = createUIStore();
  store.dispatch(actions.setFeatureToggle({ isOcpProjectStorageToggleEnabled: true }));
  expect(FeatureToggleSelectors.selectIsOcpProjectStorageToggleEnabled(store.getState())).toBe(true);
});

test('ROS feature is enabled', async () => {
  const store = createUIStore();
  store.dispatch(actions.setFeatureToggle({ isRosToggleEnabled: true }));
  expect(FeatureToggleSelectors.selectIsRosToggleEnabled(store.getState())).toBe(true);
});
