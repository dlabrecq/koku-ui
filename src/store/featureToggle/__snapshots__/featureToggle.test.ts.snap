// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default state 1`] = `
{
  "hasFeatureToggle": false,
  "isClusterInfoToggleEnabled": false,
  "isDebugToggleEnabled": false,
  "isExportsToggleEnabled": false,
  "isFinsightsToggleEnabled": false,
  "isIbmToggleEnabled": false,
  "isRosToggleEnabled": false,
  "isSettingsPlatformToggleEnabled": false,
  "isTagMappingToggleEnabled": false,
}
`;
